frontend/src/components/CategoryItem.tsx

```diff
@@ -1,12 +1,31 @@
-import {Category} from "../models/Category";
+import {Category, IDeleteCategory} from "../models/Category";
 import PaymentController from "../controllers/PaymentController";
 import {ITokenConfig} from "../models/Connection";
 import {useEffect, useState} from "react";
 import {Payment} from "../models/IPayment";
 import Payments from "./Payments";
 
-export default function CategoryItem(props: { category: Category, config: ITokenConfig }) {
-    const {category, config} = props
+const mapToCategoryItem = (category: Category, deleteCategory: IDeleteCategory) => {
+    return (
+        <div className={"categoryItem"}>
+            <h1>{category.categoryName}</h1>
+            <h2>{category.paymentSum + "€"}</h2>
+            <input type="button" value={"Remove"} onClick={() => deleteCategory(category.categoryID)}/>
+        </div>
+    )
+}
+
+const mapPayments = (payments: Payment[]) => {
+    return (
+        <div className={"payments"}>
+            <h1>Payments</h1>
+            <Payments payments={payments}/>
+        </div>
+    )
+}
+
```

Comment:
I would leave this were it was before or move it to another file. I dont like having multiple components in one file